nailgun/entities.py

```diff
@@ -544,6 +544,75 @@ def create_payload(self):
         }
 
 
+class DiscoveredHosts(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntityUpdateMixin):
+    """A representation of a Foreman Discovered Host entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(required=True),
+            'ip': entity_fields.IPAddressField(required=True),
+            'mac': entity_fields.MACAddressField(required=True),
+        }
+        self._meta = {
+            'api_path': '/api/v2/discovered_hosts',
+            'server_modes': ('sat'),
+        }
+        super(DiscoveredHosts, self).__init__(server_config, **kwargs)
+
+    def path(self, which=None):
+        """Extend ``nailgun.entity_mixins.Entity.path``.
+
+        The format of the returned path depends on the value of ``which``:
+
+        facts
+            /discovered_hosts/facts
+
+        ``super`` is called otherwise.
+
+        """
+        if which == 'facts':
+            return '{0}/facts'.format(
```

Comment:
nitpick: change to 
```python
        if which == 'facts':
            return '{0}/{1}'.format(
                super(DiscoveredHosts, self).path(which='base'),
                which
            )
```

hardcoding `'{0}/facts'` path will work in this case, but if you'd like to add some more paths in future, you'll have a lot of `if ... return` statements, 1 per each path. 
in the proposed approach you'll need to update only condition, e.g. `if which == 'foo' or which == 'bar'` and everything will work fine without any extra code.